pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -304,17 +304,18 @@ pub async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
     // Debug-log the list of candidates
     let now = SystemTime::now();
     for (i, (partition, candidate)) in candidates.iter().enumerate() {
+        let desc = candidate.layer.layer_desc();
```

Comment:
Very nice :+1:

Reply:
Sad this is that this is basically dead code... and it might be now more expensive than doing this in a `debug!` because on prod debug is never enabled so the formatting would had never been done, but let's hope rustc/llvm decide to move this method call *after* the debug interest check :)

Reply:
Good point. It's a virtual call, as the layer is a `Arc<dyn PersistentLayer>`, which can have arbitrary side effects, so I think the optimizer will not move after the check for debug, However, it should be a cheap operation so fine to do.